lib/livebook/notebook/explore/vm_introspection.livemd

```diff
@@ -0,0 +1,268 @@
+# Fun with VM introspection
+
+## Introduction
+
+In this notebook we manually establish connection to a running node,
+and then we try to retrieve and plot some interesting information
+about the system.
+
+## Setup
+
+We are definitely gonna plot some data in this notebook,
+so let's add `:vega_lite` and `:kino` for that.
+
+```elixir
+Mix.install([
+  {:vega_lite, "~> 0.1.0"},
+  {:kino, "~> 0.1.0"}
+])
+```
+
+```elixir
+alias VegaLite, as: Vl
+```
+
+## Connecting to a remote node
+
+This time we will connect to a running node manually instead of
+injecting Livebook there, because we may not want to run `Mix.install/1`
+inside it. This will also give you a better idea of how you can
+inspect a remote node in general.
+
+The first thing we need is an actually running node,
+if you don't have one at hand, feel free to start one with:
+
+`iex --sname cat --cookie meow`
+
+Next, specify the node you want to connect to below
+with the corresponding cookie:
+
+<!-- livebook:{"livebook_object":"cell_input","name":"Node","type":"text","value":"cat@your_hostname"} -->
```

Comment:
@josevalim is it more intuitive to have `cat@your_hostname` or leave the field empty by default?

Reply:
I think we need to explain what the hostname actually is. Maybe we should tell them to do `IO.puts node()` on IEx and have them copy and paste the result here? Also, the `--sname` setup is not going to work on Docker, will it? 🤔  For sure it won't work if they started with `--name`.

Reply:
If someone runs with longnames, they should easily figure out that the remote needs to be running with longnames too.

Good point about Docker, but it's isolated by design so I think neither will work unless you start the node inside the container too. Otherwise it's actually challenging, because either you don't have Elixir locally, and that's why you run with Docker, or you have Elixir locally, but then you have local epmd, and the container has its own, so connecting these properly requires a specific setup. We can simply add a disclaimer that the node needs to be discoverable and mention Docker specifically, any other ideas?

Reply:
@jonatanklosko the other option is to do the approach I did in the portal livebook, which is to ask the users to start another livebook and connect to that. And then we mention they could also use `iex --sname foo --cookie bar` or any Elixir node started in production via `mix release`.

Reply:
Ah right, this sounds like a good solution!

Reply:
Node started this way is gonna be visible in LiveDashboard though, which means the further reasoning kinda confusing :thinking:

Reply:
Perhaps we could stick to IEx, but add a note that alternatively they can start another notebook?